scripts/standby_node_health_check/standby_node_health_check.py

```diff
@@ -0,0 +1,401 @@
+#
+# Copyright © 2013-2024 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# This software product is a proprietary product of Nvidia Corporation and its affiliates
+# (the "Company") and all right, title, and interest in and to the software
+# product, including all associated intellectual property rights, are and
+# shall remain exclusively with the Company.
+#
+# This software product is governed by the End User License Agreement
+# provided with the software product.
+#
+
+import re
+import subprocess
+import argparse
+import sys
+
+
+UFM_HA_CLUSTER_COMMAND = "ufm_ha_cluster {}"
+SYSTEMCTL_IS_ACTIVE_COMMAND = "systemctl is-active {}.service"
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Checking if standby node is ready.")
+
+    parser.add_argument(
+        "--fabric-interfaces",
+        nargs="+",
+        required=True,
+        help="Specify one or more fabric interfaces (at least one is required)",
+    )
+
+    parser.add_argument(
+        "--mngmnt-interfaces",
+        nargs="+",
+        required=True,
+        help="Specify one or more management interfaces (at least one is required)",
+    )
+
+    parser.add_argument("--version", action="version", version="%(prog)s 1.0.0")
+
+    args = parser.parse_args()
+    return args
+
+
+class StandbyNodeHealthChecker:
+    def __init__(self, fabric_interfaces, mngmnt_interfaces):
+        self._fabric_interfaces = fabric_interfaces
+        self._mngmnt_interfaces = mngmnt_interfaces
+        self._ufm_ha_status_string = ""
+
+    @staticmethod
+    def _run_command(command: str):
+        try:
+            result = subprocess.run(
+                command,
+                shell=True,
+                stdout=subprocess.PIPE,
+                universal_newlines=True,
+                check=False,
+            )
+            return result.returncode, result.stdout.replace("\t", "").strip()
+        except Exception as e:
+            return -1, "", str(e)
+
+    def _check_ib_interfaces(self):
+        result = True
+        ib_interfaces_status = self._run_and_parse_ibstat()
+        ib_to_ml_map = self._get_ib_to_ml_map()
+        for ib_interface in self._fabric_interfaces:
+            ib_interface_to_validate = ib_interface
+            if not ib_interface.startswith("mlx"):
+                ib_interface_to_validate = ib_to_ml_map.get(ib_interface, ib_interface)
+            if ib_interface_to_validate not in ib_interfaces_status:
+                print(f"{ib_interface} is not in the list of IB interfaces")
+                result = False
+            elif not self._check_ib_interface(
+                ib_interface_to_validate, ib_interfaces_status
+            ):
+                print(
+                    f"IB interface {ib_interface} is not active "
```

Comment:
I think we should add a severity for all the messages - 
in that case: if only one fabric interface is down, it should be a warning (as UFM can still found an IB interface to work with) but if all given IB interfaces are down that's a critical issue.

Reply:
Replying to all the places where you talked about the severity - When we talked about this script, we agreed that for now, it would only use normal prints; this is why it was not part of this draft.

If we would like to add severity to all messages (just as in a logger), we should "just" create a logger and use it (at first stage, it can be to only print to the screen.

What do you think? @kedeme @kobibar